### PR TITLE
Add docstrings for magic methods in Bio.Restriction.Restriction

### PR DESCRIPTION
**Pull request by MarkusPiotrowski**
This pull request addresses issue #2116.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/Restriction/Restriction.py [hunk 1]
```diff
@@ -155,8 +155,8 @@ def __init__(self, seq, linear=True):
         """FormattedSeq(seq, [linear=True])-> new FormattedSeq.
```
Comment (peterjc) [inline Bio/Restriction/Restriction.py]: We should fix these docstrings which are one line summaries of the arguments and return types counter to PEP257, but that's a separate issue.

File: Bio/Restriction/Restriction.py [hunk 2]
```diff
@@ -232,6 +238,18 @@ def finditer(self, pattern, size):
         return [(i.start(), i.group) for i in re.finditer(pattern, data)]
 
     def __getitem__(self, i):
+        """Return substring of ``FormattedSeq``.
+
+        The class of the returned object is the class of the respective
+        sequence. Note that due to the leading space, indexing is 1-based::
```
Comment (peterjc) [inline Bio/Restriction/Restriction.py]: You've got a double colon meaning an indented literal block should follow - but its a doctest. This explains the flake8 error:

```
flake8
./Bio/Restriction/Restriction.py:250:1: RST399 Inconsistent literal block quoting.
ERROR: InvocationError for command /home/travis/build/biopython/biopython/.tox/style/bin/flake8 (exited with code 1)
```

Just remove the extra colon.

File: Bio/Restriction/Restriction.py (hunk 1 above)
Comment (MarkusPiotrowski) [reply Bio/Restriction/Restriction.py]: I changed them (only found two left).

File: Bio/Restriction/Restriction.py (hunk 2 above)
Comment (MarkusPiotrowski) [reply Bio/Restriction/Restriction.py]: Yes.

**Pull request merged**